ehealthid/src/main/java/com/oviva/ehealthid/auth/internal/steps/TrustedSectoralIdpStepImpl.java

```diff
@@ -69,7 +73,9 @@ public IdTokenJWS exchangeSectoralIdpCode(@NonNull String code, @NonNull String
 
       var signedJws = jweObject.getPayload().toJWSObject();
 
-      if (!JwsVerifier.verify(trustedIdpEntityStatement.body().jwks(), signedJws)) {
+      var idpSigningKeys =
+          federationMasterClient.resolveOpenIdProviderJwks(trustedIdpEntityStatement);
+      if (!JwsVerifier.verify(idpSigningKeys, signedJws)) {
         throw AuthExceptions.badIdTokenSignature(trustedIdpEntityStatement.body().sub());
       }
```

Comment:
This was the missing bit. The federation keys do not have to match the OpenID keys.